tests/e2e/nvidia-container-toolkit_test.go

```diff
@@ -257,4 +257,32 @@ var _ = Describe("docker", Ordered, ContinueOnFailure, func() {
 			Expect(libs).To(ContainElements([]string{"libcuda.so", "libcuda.so.1"}))
 		})
 	})
+
+	When("A container tries to create firmware files on the host", Ordered, func() {
+		BeforeAll(func(ctx context.Context) {
+			_, _, err := runner.Run("docker pull ubuntu")
+			Expect(err).ToNot(HaveOccurred())
+
+			_, _, err = runner.Run(`docker build -t firmware-test --build-arg RM_VERSION="$(basename $(ls -d /lib/firmware/nvidia/*.*))" --build-arg CURRENT_DIR="$(pwd)" - <<EOF
+FROM ubuntu
+RUN mkdir -p /lib/firmware/nvidia/
+ARG RM_VERSION
+ARG CURRENT_DIR
+RUN ln -s /../../../../../../../../\$CURRENT_DIR /lib/firmware/nvidia/\$RM_VERSION
+EOF`)
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		It("should not fail when using CDI", func(ctx context.Context) {
+			output, _, err := runner.Run("docker run --rm --runtime=nvidia --gpus=all firmware-test")
+			Expect(err).ToNot(HaveOccurred())
+			Expect(output).To(BeEmpty())
```

Comment:
Let's update the tests to actually check that no files are created. Using an affected version of the toolkit we see:
```
$ docker run --rm -ti --gpus=all --runtime=nvidia firmware-test true
$ ls
gsp_ga10x.bin  gsp_tu10x.bin
```
So we would ensure that `ls gsp_*.bin` is empty.